hyperspy/roi.py

```diff
@@ -173,6 +190,11 @@ def __call__(self, signal, out=None, axes=None):
               space can fit the right number of axis, and use that if it
               fits. If not, it will try the signal space.
         """
+        if t.Undefined in list(self):
+            raise AttributeError(
+        "Some ROI parameters have not yet been set. "
```

Comment:
Since this error message is used several times, I would define it in this module and reuse it. I am not sure this is a `AttributeError`, because the attribute exists and doesn't fail, however its value is not compatible, so I would think, this is more a `ValueError`.
https://docs.python.org/3/library/exceptions.html#AttributeError